Parsely/parsely-android
# Support `siteId` per tracking request

Closes: #115 

### Description

This PR introduces a possibility for SDK users to provide a custom `siteId` whenever they record a pixel/track event.

This will also align to iOS, which supports this feature already

https://github.com/Parsely/AnalyticsSDK-iOS/blob/9b9db5478996582fdbd07d5aa4a2da0134f9ffe7/Sources/ParselyTracker.swift#L47-L48

#### Design

I've decided to propose more verbose API and introduce `SiteIdSource` instead of a `String` which is empty by default. I think this way it's easier for SDK users to understand the behavior.

#### Backwards compatibility

Because we cannot introduce `@JvmOverloads` on interface methods, this change will be breaking for Java users. But I think this is acceptable - I plan to release `4.1.0` after this (not bumping major version). In the case of Java users, addressing breaking changes will be as easy as providing `SiteIdSource.Default.INSTANCE` in tracking methods.

### Test
This PR introduces unit tests and functional test case for engagement manager case. We can also test this with example app:

1. Run the example app
2. Provide some value in "Custom site id" edit text
3. Tap on "Track URL", then press "Start engagement"
4. Observe in logs that after ~15 seconds you should see logs that would be sent to the backend. Make sure they have `idsite` equal to the custom id you provided in step 2.
